src/app/@ansyn/plugins/openlayers/maps/open-layers-map/openlayers-map/openlayers-map.ts

```diff
@@ -32,24 +45,56 @@ export enum StaticGroupsKeys {
 // @dynamic
 @ImageryMap({
 	mapType: OpenlayersMapName,
-	deps: [OpenLayersProjectionService, CoreConfig],
+	deps: [HttpClient, OpenLayersProjectionService, CoreConfig, TilesLoadingConfig],
 	defaultMapSource: 'BING'
 })
 export class OpenLayersMap extends BaseImageryMap<OLMap> {
 	static groupsKeys = StaticGroupsKeys;
 	static groupLayers = new Map<StaticGroupsKeys, Group>(Object.values(StaticGroupsKeys).map((key) => [key, new Group()]) as any);
 	private showGroups = new Map<StaticGroupsKeys, boolean>();
 	private _mapObject: OLMap;
+	private _backgroundMapObject: OLMap;
+	private _backgroundMapParams: object;
+
 	private _moveEndListener: () => void;
 	private olGeoJSON: OLGeoJSON = new OLGeoJSON();
 	private _mapLayers = [];
 	public isValidPosition;
-	public shadowElement = null;
+	public shadowNorthElement = null;
+	private isLoading$: Subject<boolean> = new Subject();
+
+	private monitor: OpenLayersMonitor = new OpenLayersMonitor(
+		this.tilesLoadProgressEventEmitter,
+		this.tilesLoadErrorEventEmitter,
+		this.http
+	);
+
+	signalWhenTilesLoadingEnds() {
+		this.isLoading$.next(true);
+		this.tilesLoadProgressEventEmitter.pipe(
+			filter((payload: IMapProgress) => {
+				return payload.progress === 100;
+			}),
+			debounceTime(this.tilesLoadingConfig.debounceTimeInMs), // Adding debounce, to compensate for strange multiple loads when reading tiles from the browser cache (e.g. after browser refresh)
```

Comment:
this.tilesLoadingConfig.(debounceTimeInMs || timeoutInMs) should have hard coded value when not providing a value from config

Reply:
done